src/dso/_mv.py

```diff
@@ -0,0 +1,308 @@
+        exit(1)
+
+    # determine how many digits are at the end of source.name starting from the last character
+    source_prefix_number_of_digits = _get_n_numeric(source_prefix)
+
+    if source_prefix_number_of_digits == 0:
+        log.error(f"[red]Prefix '{source_prefix}' must end with at least one digit.")
+        exit(1)
+
+    if source_prefix_number_of_digits != target_prefix_number_of_digits:
+        log.error(f"[red]Prefix '{source_prefix}' and '{target_prefix}' must have the same number of digits.")
+        exit(1)
+
+    # get the numeric indeces and calculate the offset
+    source_prefix_index = int(source_prefix[-source_prefix_number_of_digits:])
+    target_prefix_index = int(target_prefix[-target_prefix_number_of_digits:])
+
+    offset = target_prefix_index - source_prefix_index
+
+    # get the stem of the prefixes
+    source_prefix_stem = source_prefix[:-source_prefix_number_of_digits]
+    target_prefix_stem = target_prefix[:-target_prefix_number_of_digits]
+
+    # Create the regex pattern for matching source files
+    source_pattern = rf"^{escape(source_prefix_stem)}(\d{{{source_prefix_number_of_digits}}})(.+)"
+
+    # Find all files/folders in parent_dir matching the pattern and prepare list with
+    # source and target info
+    source_target_list = []
+    pattern = re.compile(source_pattern)
+    for entry in parent_dir.iterdir():
+        m = pattern.match(entry.name)
+        if m:
+            new_target_index = int(m.group(1))
+            if not new_target_index < source_prefix_index:
+                source_target_list.append(
+                    {
+                        "source_prefix": source_prefix_stem,
+                        "source_index": m.group(1),
+                        "source_name": m.group(2),
+                        "source_path": entry,
+                        "target_index": int(m.group(1)) + offset,
+                        "target_prefix": target_prefix_stem,
+                        "target_index_length": target_prefix_number_of_digits,
+                        "target_path": parent_dir
+                        / (
+                            target_prefix_stem
+                            + str(int(m.group(1)) + offset).zfill(target_prefix_number_of_digits)
+                            + m.group(2)
+                        ),
+                    }
+                )
+
+    # Sort source_target_list by 'target_index' in decreasing order for work through the list to avoid collisions
+    source_target_list.sort(key=lambda x: x["target_index"], reverse=True)
+
+    # rename all sources, if the source is a directory with a dvc.yaml file, then use `dso mv`,
+    # otherwise just rename the directory
+    for item in source_target_list:
+        log.info(f"[yellow]Renaming {item['source_path'].name} to {item['target_path'].name} in {parent_dir}")
+
+        if (item["target_path"] / "dvc.yaml").exists():
```

Comment:
```suggestion
        if (item["source_path"] / "dvc.yaml").exists():
```

This should be source_path, no? 

When I tried `mv --increment-prefix`, it didn't update the stages, just moved them, probably because of that.